### PR TITLE
Fix grammar in linux-initialization-1.

### PR DESCRIPTION
**Pull request by JDuchniewicz**
More PRs will follow for next sections. Very helpful guide - adding some touches on style and grammar to make it easier to read.

### CODE REVIEW

**Comment by luohuang** [comment]
这是来自QQ邮箱的自动回复邮件。
 
您好，您的邮件我已经收到。
 
谢谢！

**Comment by JDuchniewicz** [comment]
Any plans to merge it @mudongliang? I saw you are maintaining this repo currently - I update stylistics and minor errors as I am reading through it. Hope it helps :)

**Comment by mudongliang** [comment]
Hi, your PR is a little long, I need some time to review them. Please be patient.

File: Initialization/linux-initialization-1.md [hunk 1]
```diff
@@ -4,15 +4,15 @@ Kernel initialization. Part 1.
...
...nitialization process of the Linux kernel. After the image of the Linux kernel is decompressed and placed in a correct place in memory, it starts to work. All previous parts describe the work of the Linux kernel setup code which performs preparation before the first bytes of the Linux kernel code will be executed. From now on we are in the kernel and all parts of this chapter will be devoted to the initialization process of the kernel before it launches the process with the [pid](https://en.wikipedia.org/wiki/Process_identifier) `1`. There are many things to be done before the kernel starts first `init` process. Hopefully we will see all of the preparations before kernel starts throughout this big chapter. We will start from the kernel entry point, that is located in the [arch/x86/kernel/head_64.S](https://github.com/torvalds/linux/blob/master/arch/x86/kernel/head_64.S) and will move further and further. We will see the initial preparations such as early page tables initialization, switching to a new descriptor in kernel space and many many more, before we see the `start_kernel` function from the [init/main.c](https://github.com/torvalds/linux/blob/master/init/main.c) being called.
```
Comment (mudongliang) [inline Initialization/linux-initialization-1.md:7]: starts first `init` process. -> starts the first `init` process

Comment (mudongliang) [inline Initialization/linux-initialization-1.md:7]: before kernel starts -> before the kernel starts  

Maybe you can try `in the kernel initialization`.

Comment (mudongliang) [inline Initialization/linux-initialization-1.md:7]: We will see the initial preparations such as early page tables initialization, switching to a new descriptor in kernel space and many many more, before we see the `start_kernel` function from the [init/main.c](https://github.com/torvalds/linux/blob/master/init/main.c) being called
->
We will see the initial preparations, such as, early page tables initialization, switching to a new descriptor in kernel space and many many more, before the function `start_kernel` from [init/main.c](https://github.com/torvalds/linux/blob/master/init/main.c) is called.

File: Initialization/linux-initialization-1.md [hunk 2]
```diff
@@ -4,15 +4,15 @@ Kernel initialization. Part 1.
...
 
 ```assembly
 jmp	*%rax
 ```
 
-At this moment the `rax` register contains address of the Linux kernel entry point which was obtained as a result of the call of the `decompress_kernel` function from the [arch/x86/boot/compressed/misc.c](https://github.com/torvalds/linux/blob/master/arch/x86/boot/compressed/misc.c) source code file. So, our last instruction in the kernel setup code is a jump on the kernel entry point. We already know where the entry point of the Linux kernel is defined, so we are able to start to learn what Linux kernel does after the start.
+At this moment the `rax` register contains address of the Linux kernel entry point which was obtained as a result of the call of the `decompress_kernel` function from the [arch/x86/boot/compressed/misc.c](https://github.com/torvalds/linux/blob/master/arch/x86/boot/compressed/misc.c) source code file. So, our last instruction in the kernel setup code is a jump to the kernel entry point. We already know where the entry point of the Linux kernel is defined, so we are able to begin learning what Linux kernel does after the start.
```
Comment (mudongliang) [inline Initialization/linux-initialization-1.md:15]: "start" to "begin" is not necessary

File: Initialization/linux-initialization-1.md [hunk 3]
```diff
@@ -47,10 +47,10 @@ We can see definition of this section in the [arch/x86/kernel/vmlinux.lds.S](htt
 } :text = 0x9090
 ```
 
-The ADDR keyword above returns the absolute address (here means virtual address) of the named section. The AT keyword above specifies the load address (here means physical address) of the section. The full syntax of section definition is defined in the [Using ld The GNU linker](https://ftp.gnu.org/old-gnu/Manuals/ld-2.9.1/html_node/ld_21.html). 
+The ADDR keyword above returns the absolute address (here meaning the virtual address) of the named section. The AT keyword above specifies the load address (here meaning the physical address) of the section. The full syntax of section definition is defined in the [Using ld The GNU linker](https://ftp.gnu.org/old-gnu/Manuals/ld-2.9.1/html_node/ld_21.html).
```
Comment (mudongliang) [inline Initialization/linux-initialization-1.md:50]: I think you can modify `here means` to `i.e.,`

**Comment by mudongliang** [comment]
A small suggestion: please split the changes of the whole section into several smaller PRs. It is easier for people to review.

**Comment by JDuchniewicz** [comment]
I think that splitting it in section by section PRs will be even harder to review as there will be a lot of them since every chapter contains 5-6 sections.

**Comment by mudongliang** [comment]
> I think that splitting it in section by section PRs will be even harder to review as there will be a lot of them since every chapter contains 5-6 sections.

Yeah, you can balance the trade-off

File: Initialization/linux-initialization-1.md (hunk 2 above)
Comment (mjbear) [reply Initialization/linux-initialization-1.md:15]: > "start" to "begin" is not necessary

@mudongliang
Are you suggesting the wording be kept as is?
Or are you indicating it be simplified?

> We already know where the entry point of the Linux kernel is defined, so we are able to begin learning what Linux kernel does after the start.

We already know where the entry point of the Linux kernel is defined, so ***next*** we ~are able to~ begin learning what ***the*** Linux kernel does after the start.

File: Initialization/linux-initialization-1.md [hunk 4]
```diff
@@ -440,7 +440,7 @@ The `THREAD_SIZE` macro is defined in the [arch/x86/include/asm/page_64_types.h]
...
+We consider the case when the [kasan](https://github.com/torvalds/linux/blob/master/Documentation/dev-tools/kasan.rst) is disabled and the `PAGE_SIZE` is `4096` bytes. So the `THREAD_SIZE` will expand to `16` kilobytes and represent size of the stack of a thread. Why is it a `thread`? You may already know that each [process](https://en.wikipedia.org/wiki/Process_%28computing%29) may have a [parent processes](https://en.wikipedia.org/wiki/Parent_process) and [child processes](https://en.wikipedia.org/wiki/Child_process). Actually, a parent process and child process differ in stack. A new kernel stack is allocated for a new process. In the Linux kernel this stack is represented by the [union](https://en.wikipedia.org/wiki/Union_type#C.2FC.2B.2B) with the `thread_info` structure.
```
Comment (mjbear) [inline Initialization/linux-initialization-1.md:443]: Removing the plural from the link `parent processes` (grammatically and to match the hyperlink as well).

```suggestion
We consider the case when the [kasan](https://github.com/torvalds/linux/blob/master/Documentation/dev-tools/kasan.rst) is disabled and the `PAGE_SIZE` is `4096` bytes. So the `THREAD_SIZE` will expand to `16` kilobytes and represent size of the stack of a thread. Why is it a `thread`? You may already know that each [process](https://en.wikipedia.org/wiki/Process_%28computing%29) may have a [parent process](https://en.wikipedia.org/wiki/Parent_process) and [child processes](https://en.wikipedia.org/wiki/Child_process). Actually, a parent process and child process differ in stack. A new kernel stack is allocated for a new process. In the Linux kernel this stack is represented by the [union](https://en.wikipedia.org/wiki/Union_type#C.2FC.2B.2B) with the `thread_info` structure.
```

File: Initialization/linux-initialization-1.md [hunk 5]
```diff
@@ -616,12 +616,12 @@ asmlinkage __visible void __init x86_64_start_kernel(char * real_mode_data)
 }
 ```
 
-It has one argument is a `real_mode_data` (remember that we passed address of the real mode data to the `rdi` register previously).
+It has a one argument is a `real_mode_data` (remember that we passed address of the real mode data to the `rdi` register previously).
```
Comment (mjbear) [inline Initialization/linux-initialization-1.md:619]: Maybe rephrase the sentence?

```suggestion
It has one argument named `real_mode_data` (remember that we passed address of the real mode data to the `rdi` register previously).
```

**Comment by 0xAX** [comment]
The first chapter was fully reworked in https://github.com/0xAX/linux-insides/pull/915.
